cli/pcluster/commands.py

```diff
@@ -685,6 +704,23 @@ def delete(args):
         sys.exit(0)
 
 
+def delete(args):
+    PclusterConfig.init_aws(config_file=args.config_file)
+    LOGGER.info("Deleting: %s", args.cluster_name)
+    stack_name = utils.get_stack_name(args.cluster_name)
+    stack_exists = utils.stack_exists(stack_name)
+    if args.keep_logs and not stack_exists:
+        utils.error(
+            "Stack for {} does not exist. Cannot prevent its log group from being deleted.".format(args.cluster_name)
+        )
+    elif args.keep_logs:
+        _persist_cloudwatch_log_group(args)
+    elif not stack_exists:
```

Comment:
probably you wanna check this first

Reply:
It is checked in the first conditional. The conditions in order are:
1. stack doesn't exist but the user is trying to preserve its logs
2. stack exists and the user is trying to preserve its logs
3. stack doesn't exist and the user isn't trying to preserve its logs

Reply:
Ended up moving it to the first conditional anyways.